src/Microsoft.DocAsCode.Build.Engine/Incrementals/IncrementalBuildContext.cs

```diff
@@ -592,23 +592,34 @@ private static DependencyGraph ConstructDependencyGraphFromLast(DependencyGraph
 
         private static IncrementalStatus GetBuildInfoIncrementalStatus(BuildInfo cb, BuildInfo lb)
         {
+            var details = String.Empty;
+            var canIncremental = false;
+
             if (lb == null)
             {
-                return new IncrementalStatus { CanIncremental = false, Details = "Cannot build incrementally because last build info is missing." };
+                details = "Cannot build incrementally because last build info is missing.";
+            }
+            else if (cb.DocfxVersion != lb.DocfxVersion)
+            {
+                details = $"Cannot build incrementally because docfx version changed from {lb.DocfxVersion} to {cb.DocfxVersion}.";
             }
-            if (cb.DocfxVersion != lb.DocfxVersion)
+            else if (cb.PluginHash != lb.PluginHash)
             {
-                return new IncrementalStatus { CanIncremental = false, Details = $"Cannot build incrementally because docfx version changed from {lb.DocfxVersion} to {cb.DocfxVersion}." };
+                details = "Cannot build incrementally because plugin changed.";
             }
-            if (cb.PluginHash != lb.PluginHash)
+            else if (cb.CommitFromSHA != lb.CommitToSHA)
             {
-                return new IncrementalStatus { CanIncremental = false, Details = "Cannot build incrementally because plugin changed." };
+                details = $"Cannot build incrementally because commit SHA doesn't match. Last build commit: {lb.CommitToSHA}. Current build commit base: {cb.CommitFromSHA}.";
             }
-            if (cb.CommitFromSHA != lb.CommitToSHA)
+            else
             {
-                return new IncrementalStatus { CanIncremental = false, Details = $"Cannot build incrementally because commit SHA doesn't match. Last build commit: {lb.CommitToSHA}. Current build commit base: {cb.CommitFromSHA}." };
+                canIncremental = true;
             }
-            return new IncrementalStatus { CanIncremental = true };
+
+            // Log
+            Logger.LogInfo($"Build strategy", code: canIncremental ? WarningCodes.Build.IsIncrementalBuild : WarningCodes.Build.IsFullBuild);
```

Comment:
>LogInfo [](start = 19, length = 7)

print strategy in message itself

Reply:
Added.

Reply:
also add why

---
In reply to: [200141550](https://github.com/dotnet/docfx/pull/3002#discussion_r200141550) [](ancestors = 200141550)

Reply:
You means more `message` information?